src/platform/ESP32/WiFiDnssdImpl.cpp

```diff
@@ -323,12 +324,22 @@ static CHIP_ERROR OnBrowseDone(BrowseContext * ctx)
                 ctx->mService[servicesIndex].mAddressType   = MapAddressType(currentResult->ip_protocol);
                 ctx->mService[servicesIndex].mTransportType = ctx->mAddressType;
                 ctx->mService[servicesIndex].mPort          = currentResult->port;
-                ctx->mService[servicesIndex].mInterface     = ctx->mInterfaceId;
                 ctx->mService[servicesIndex].mTextEntries =
                     GetTextEntry(currentResult->txt, currentResult->txt_value_len, currentResult->txt_count);
                 ctx->mService[servicesIndex].mTextEntrySize = currentResult->txt_count;
                 ctx->mService[servicesIndex].mSubTypes      = NULL;
                 ctx->mService[servicesIndex].mSubTypeSize   = 0;
+                if (ctx->mInterfaceId == chip::Inet::InterfaceId::Null())
+                {
+                    // If the InterfaceId in the context is Null, we will use the Station netif by default.
+                    struct netif * lwip_netif =
+                        reinterpret_cast<struct netif *>(esp_netif_get_netif_impl(esp_netif_get_handle_from_ifkey("WIFI_STA_DEF")));
```

Comment:
May cause some issue to Ethenet devices. @PSONALl Please have a check.

Reply:
This file is only for Wi-Fi devices so I don't think is will cause issue to Ethernet devices.